Execution failed for task ':transformAtomicfuClasses'. NestHost requires ASM7

```
plugins {
    kotlin("jvm") version "1.5.30"
    java
}

buildscript {
    dependencies {
        classpath("org.jetbrains.kotlinx:atomicfu-gradle-plugin:0.16.3")
    }
}
apply(plugin = "kotlinx-atomicfu")

group = "org.akhsaul"
version = "1.0-SNAPSHOT"
java.targetCompatibility = JavaVersion.VERSION_15
val kotlinVersion = "1.5.30"
val okhttp3Version = "5.0.0-alpha.2"
val coroutineVersion = "1.5.1-native-mt"

repositories {
    google()
    mavenCentral()
    gradlePluginPortal()
    maven {
        isAllowInsecureProtocol = true
        url = uri("http://4thline.org/m2")
    }
    maven {
        url = uri("https://jitpack.io")
    }
}

dependencies {
    implementation("org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion")
    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutineVersion")
    implementation("com.squareup.okhttp3:okhttp:$okhttp3Version")
    implementation("com.squareup.okhttp3:logging-interceptor:$okhttp3Version")
    implementation("com.squareup.okhttp3:okhttp-urlconnection:$okhttp3Version")
    implementation("com.squareup.okhttp3:okhttp-dnsoverhttps:$okhttp3Version")
    implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.13.+")
    implementation("commons-io:commons-io:2.11.0")
    testImplementation("org.jetbrains.kotlin:kotlin-test-junit5:$kotlinVersion")
    testImplementation("org.junit.jupiter:junit-jupiter-api:5.7.2")
    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.7.2")
}

tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
    kotlinOptions {
        freeCompilerArgs = listOf(
            "-Xjsr305=strict",
            "-Xextended-compiler-checks"
        )
        jvmTarget = "15"
    }
    kotlin.sourceSets.all {
        languageSettings {
            optIn("kotlin.Experimental")
            optIn("kotlin.ExperimentalStdlibApi")
            optIn("kotlin.io.path.ExperimentalPathApi")
            optIn("kotlin.contracts.ExperimentalContracts")
            optIn("kotlin.RequiresOptIn")
        }
    }
}

tasks.withType<Test> {
    useJUnitPlatform()
}
```

when i try `Executing task 'build'`, gradle throw error:

```
14:17:43: Executing task 'build'...

> Task :compileKotlin UP-TO-DATE
> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :transformAtomicfuClasses FAILED
4 actionable tasks: 1 executed, 3 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':transformAtomicfuClasses'.
> NestHost requires ASM7

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 459ms
14:17:44: Task execution finished 'build'.
```

I just encountered more or less the same error when using `atomicfu` in a project that includes a JPMS module descriptor.
ASM is unable to process the `module-info.class` file and fails with this exception:
```

Caused by: java.lang.UnsupportedOperationException: Module requires ASM6
	at org.objectweb.asm.ClassVisitor.visitModule(ClassVisitor.java:145)
	at org.objectweb.asm.ClassReader.readModuleAttributes(ClassReader.java:758)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:557)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:401)
	at kotlinx.atomicfu.transformer.AtomicFUTransformer.analyzeFileForFields(AtomicFUTransformer.kt:234)
	at kotlinx.atomicfu.transformer.AtomicFUTransformer.analyzeFilesForFields(AtomicFUTransformer.kt:227)
	at kotlinx.atomicfu.transformer.AtomicFUTransformer.transform(AtomicFUTransformer.kt:196)
	at kotlinx.atomicfu.plugin.gradle.AtomicFUTransformTask.transform(AtomicFUGradlePlugin.kt:427)
	at jdk.internal.reflect.GeneratedMethodAccessor654.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        ...truncated...


* Get more help at https://help.gradle.org

BUILD FAILED in 2s
09:22:53: Task execution finished 'assemble --stacktrace'.
```

--- a/atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt
+++ b/atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt
@@ -126,7 +126,7 @@ private inline fun code(mv: MethodVisitor, block: InstructionAdapter.() -> Unit)
 }
 
 private inline fun insns(block: InstructionAdapter.() -> Unit): InsnList {
-    val node = MethodNode(ASM5)
+    val node = MethodNode(ASM9)
     block(InstructionAdapter(node))
     return node.instructions
 }
@@ -280,7 +280,7 @@ class AtomicFUTransformer(
         return cw.toByteArray() // write transformed bytes
     }
 
-    private abstract inner class CV(cv: ClassVisitor?) : ClassVisitor(ASM5, cv) {
+    private abstract inner class CV(cv: ClassVisitor?) : ClassVisitor(ASM9, cv) {
         lateinit var className: String
 
         override fun visit(
@@ -348,7 +348,7 @@ class AtomicFUTransformer(
     private inner class AccessorCollectorMV(
         private val className: String,
         access: Int, name: String, desc: String, signature: String?, exceptions: Array<out String>?
-    ) : MethodNode(ASM5, access, name, desc, signature, exceptions) {
+    ) : MethodNode(ASM9, access, name, desc, signature, exceptions) {
         override fun visitEnd() {
             val insns = instructions.listUseful(4)
             if (insns.size == 3 &&
@@ -398,7 +398,7 @@ class AtomicFUTransformer(
 
     private inner class DelegateFieldsCollectorMV(
             access: Int, name: String, desc: String, signature: String?, exceptions: Array<out String>?
-    ) : MethodNode(ASM5, access, name, desc, signature, exceptions) {
+    ) : MethodNode(ASM9, access, name, desc, signature, exceptions) {
         override fun visitEnd() {
             // register delegate field and the corresponding original atomic field
             // getfield a: *Atomic
@@ -460,7 +460,7 @@ class AtomicFUTransformer(
         private var originalClinit: MethodNode? = null
         private var newClinit: MethodNode? = null
 
-        private fun newClinit() = MethodNode(ASM5, ACC_STATIC, "<clinit>", "()V", null, null)
+        private fun newClinit() = MethodNode(ASM9, ACC_STATIC, "<clinit>", "()V", null, null)
         fun getOrCreateNewClinit(): MethodNode = newClinit ?: newClinit().also { newClinit = it }
 
         override fun visitSource(source: String?, debug: String?) {
@@ -607,7 +607,7 @@ class AtomicFUTransformer(
             val superMV = if (name == "<clinit>" && desc == "()V") {
                 if (access and ACC_STATIC == 0) abort("<clinit> method not marked as static")
                 // defer writing class initialization method
-                val node = MethodNode(ASM5, access, name, desc, signature, exceptions)
+                val node = MethodNode(ASM9, access, name, desc, signature, exceptions)
                 if (originalClinit != null) abort("Multiple <clinit> methods found")
                 originalClinit = node
                 node
@@ -673,7 +673,7 @@ class AtomicFUTransformer(
         private val packageName: String,
         private val vh: Boolean,
         private val analyzePhase2: Boolean // true in Phase 2 when we are analyzing file for refs (not transforming yet)
-    ) : MethodNode(ASM5, access, name, desc, signature, exceptions) {
+    ) : MethodNode(ASM9, access, name, desc, signature, exceptions) {
         init {
             this.mv = mv
         }

